File: fontend/smart_dine/lib/models/reservation.dart
class Reservation {
  final int? id;
  final int branchId;
  final String customerName;
  final String customerEmail;
  final DateTime? reservedTime;
  final int numberOfGuests;
  final DateTime? reservedDay;
  final int statusId;
  final String? note;

  Reservation({
    this.id,
    required this.branchId,
    required this.customerName,
    required this.customerEmail,
    this.reservedTime,
    required this.numberOfGuests,
    this.reservedDay,
    required this.statusId,
    this.note,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'],
      branchId: json['branchId'],
      customerName: json['customerName'],
      customerEmail: json['customerEmail'],
      reservedTime:
          json['reservedTime'] != null
              ? DateTime.parse(json['reservedTime'])
              : null,
      numberOfGuests: json['numberOfGuests'],
      reservedDay:
          json['reservedDay'] != null
              ? DateTime.parse(json['reservedDay'])
              : null,
      statusId: json['statusId'],
      note: json['note'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'branchId': branchId,
      'customerName': customerName,
      'customerEmail': customerEmail,
      'reservedTime': reservedTime?.toIso8601String(),
      'numberOfGuests': numberOfGuests,
      'reservedDay': reservedDay?.toIso8601String(),
      'statusId': statusId,
      'note': note,
    };
  }
}

File: src/main/java/com/smartdine/models/Reservation.java
package com.smartdine.models;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;

@Entity
@Table(name = "reservations")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "branch_id")
    private Integer branchId;

    @Column(name = "customer_name")
    private String customerName;

    @Column(name = "customer_email")
    private String customerEmail;

    @Column(name = "reserved_time")
    private LocalDateTime reservedTime;

    @Column(name = "number_of_guests")
    private Integer numberOfGuests;

    @Column(name = "reserved_day")
    private LocalDateTime reservedDay;

    @Column(name = "status_id")
    private Integer statusId;

    @Column(name = "note", columnDefinition = "TEXT")
    private String note;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public Reservation() {
    }

    public Reservation(Integer id, Integer branchId, String customerName, String customerEmail,
            LocalDateTime reservedTime, Integer numberOfGuests, LocalDateTime reservedDay, Integer statusId,
            String note, LocalDateTime createdAt, LocalDateTime updatedAt) {
        this.id = id;
        this.branchId = branchId;
        this.customerName = customerName;
        this.customerEmail = customerEmail;
        this.reservedTime = reservedTime;
        this.numberOfGuests = numberOfGuests;
        this.reservedDay = reservedDay;
        this.statusId = statusId;
        this.note = note;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getBranchId() {
        return branchId;
    }

    public void setBranchId(Integer branchId) {
        this.branchId = branchId;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public String getCustomerEmail() {
        return customerEmail;
    }

    public void setCustomerEmail(String customerEmail) {
        this.customerEmail = customerEmail;
    }

    public LocalDateTime getReservedTime() {
        return reservedTime;
    }

    public void setReservedTime(LocalDateTime reservedTime) {
        this.reservedTime = reservedTime;
    }

    public Integer getNumberOfGuests() {
        return numberOfGuests;
    }

    public void setNumberOfGuests(Integer numberOfGuests) {
        this.numberOfGuests = numberOfGuests;
    }

    public LocalDateTime getReservedDay() {
        return reservedDay;
    }

    public void setReservedDay(LocalDateTime reservedDay) {
        this.reservedDay = reservedDay;
    }

    public Integer getStatusId() {
        return statusId;
    }

    public void setStatusId(Integer statusId) {
        this.statusId = statusId;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    @PrePersist
    protected void onCreate() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
        if (this.updatedAt == null) {
            this.updatedAt = LocalDateTime.now();
        }
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
